spec/controllers/videos_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+describe VideosController do
+  let(:user) { Fabricate(:user) }
+  let(:video) { Fabricate(:video) }
+
+  describe 'GET #index' do
+    it 'sets the @categories variable for authenticated users' do
+      request.session['user_id'] = user.id
+
+      local_video = video
+
+      get :index
+      expect(assigns(:categories)).to eq([local_video.category])
+    end
+
+    it 'redirects to root path for unauthenticated users' do
+      local_video = video
+
+      get :index
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe 'GET #show' do
+    context 'authenticated users' do
+      before { request.session['user_id'] = user.id }
```

Comment:
Hm indeed. I wonder why I thought it would be better to make separate before blocks.